### PR TITLE
#20 set celery.base from string

### LINKED ISSUE #20: How to set celery.base
I have an abstract celery base Task for adding extra logging with our celery tasks. Like so:

```
@task(queue='email', base=LoggingTask)
def foo()
...
```
 But I can't set that in the settings, can't import it since it imports celery which causes circular import and can't use a string.

```
CELERY_EMAIL_TASK_CONFIG = {
    # we want emails to be high priority
    'queue': 'emails',
    # can also configure rate limit
    # 'rate_limit': '50/m',
    'ignore_result': True,
    # extra logging
    'base': 'celeryutils.LoggingTask',
}
```

http://celery.readthedocs.org/en/latest/userguide/tasks.html#abstract-classes

### PR DESCRIPTION
**Pull request by oppianmatt**
fixes #20

### CODE REVIEW

File: djcelery_email/tasks.py [hunk 1]
```diff
@@ -17,6 +17,15 @@
 TASK_CONFIG = {'name': 'djcelery_email_send_multiple', 'ignore_result': True}
 TASK_CONFIG.update(settings.CELERY_EMAIL_TASK_CONFIG)
 
+# import base if string to allow a base celery task
+if 'base' in TASK_CONFIG and isinstance(TASK_CONFIG['base'], str):
```
Comment (pmclanahan) [inline djcelery_email/tasks.py]: Correct me if I'm wrong, but I believe it's best to use `basestring` as the type for an `isinstance` check.

**Comment by pmclanahan** [comment]
Apart from my one note this looks wonderful! Thanks for the great PR. I'd love to see a test, but we can add that later if need be.

**Comment by oppianmatt** [comment]
You're right, it's basestring. Also went to add a test but the test runner wasn't running.

```
...
Installing collected packages: Django, pytz, billiard, amqp, anyjson, kombu, celery, six, django-appconf, pyflakes, pep8, mccabe, flake8
Successfully installed Django-1.8.4 amqp-1.4.6 anyjson-0.3.3 billiard-3.3.0.20 celery-3.1.18 django-appconf-1.0.1 flake8-2.4.1 kombu-3.0.26 mccabe-0.3.1 pep8-1.5.7 pyflakes-0.8.1 pytz-2015.4 six-1.9.0
(venv)~/S/django-celery-email (_mj_base_task ⚡☡=) cd test_project/
(venv)~/S/d/test_project (_mj_base_task ⚡☡=) ls
__init__.py     manage.py       settings.py     settings.pyc    test_runner.py  test_runner.pyc tester          urls.py         utils.py
(venv)~/S/d/test_project (_mj_base_task ⚡☡=) ./manage.py test --noinput --failfast
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/commands/test.py", line 30, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/base.py", line 381, in run_from_argv
    parser = self.create_parser(argv[0], argv[1])
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/base.py", line 354, in create_parser
    self.add_arguments(parser)
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/core/management/commands/test.py", line 52, in add_arguments
    test_runner_class = get_runner(settings, self.test_runner)
  File "/Users/matt/Source/django-celery-email/venv/lib/python2.7/site-packages/django/test/utils.py", line 152, in get_runner
    test_module = __import__(test_module_name, {}, {}, force_str(test_path[-1]))
  File "/Users/matt/Source/django-celery-email/test_project/test_runner.py", line 2, in <module>
    from django.test.simple import DjangoTestSuiteRunner
ImportError: No module named simple
```

**Comment by pmclanahan** [comment]
The way travis runs the test is `python setup.py test`. Have you tried that command?

**Comment by oppianmatt** [comment]
previously was using the manage.py in the test_project, but running python setup.py test still gives the same result:

```
(django-celery-email)~/S/django-celery-email (_mj_base_task ☡=) pip install -r requirements.txt
Downloading/unpacking Django (from -r requirements.txt (line 1))
  Downloading Django-1.8.4-py2.py3-none-any.whl (6.2MB): 6.2MB downloaded
Downloading/unpacking celery>=2.3.0 (from -r requirements.txt (line 2))
  Downloading celery-3.1.18-py2.py3-none-any.whl (515kB): 515kB downloaded
Downloading/unpacking django-appconf (from -r requirements.txt (line 3))
  Downloading django_appconf-1.0.1-py2.py3-none-any.whl
Downloading/unpacking flake8 (from -r requirements.txt (line 4))
  Downloading flake8-2.4.1-py2.py3-none-any.whl
Downloading/unpacking billiard>=3.3.0.20,<3.4 (from celery>=2.3.0->-r requirements.txt (line 2))
  Downloading billiard-3.3.0.20.tar.gz (150kB): 150kB downloaded
  Running setup.py (path:/Users/matt/.virtualenvs/django-celery-email/build/billiard/setup.py) egg_info for package billiard

    warning: no files found matching '*.py' under directory 'Lib'
Downloading/unpacking pytz>dev (from celery>=2.3.0->-r requirements.txt (line 2))
  Downloading pytz-2015.4-py2.py3-none-any.whl (475kB): 475kB downloaded
Downloading/unpacking kombu>=3.0.25,<3.1 (from celery>=2.3.0->-r requirements.txt (line 2))
  Downloading kombu-3.0.26-py2.py3-none-any.whl (238kB): 238kB downloaded
Downloading/unpacking six (from django-appconf->-r requirements.txt (line 3))
  Downloading six-1.9.0-py2.py3-none-any.whl
Downloading/unpacking mccabe>=0.2.1,<0.4 (from flake8->-r requirements.txt (line 4))
  Downloading mccabe-0.3.1-py2.py3-none-any.whl
Downloading/unpacking pep8>=1.5.7,!=1.6.0,!=1.6.1,!=1.6.2 (from flake8->-r requirements.txt (line 4))
  Downloading pep8-1.5.7-py2.py3-none-any.whl
Downloading/unpacking pyflakes>=0.8.1,<0.9 (from flake8->-r requirements.txt (line 4))
  Downloading pyflakes-0.8.1-py2.py3-none-any.whl
Downloading/unpacking amqp>=1.4.5,<2.0 (from kombu>=3.0.25,<3.1->celery>=2.3.0->-r requirements.txt (line 2))
  Downloading amqp-1.4.6-py2-none-any.whl (49kB): 49kB downloaded
Downloading/unpacking anyjson>=0.3.3 (from kombu>=3.0.25,<3.1->celery>=2.3.0->-r requirements.txt (line 2))
  Downloading anyjson-0.3.3.tar.gz
  Running setup.py (path:/Users/matt/.virtualenvs/django-celery-email/build/anyjson/setup.py) egg_info for package anyjson

Installing collected packages: Django, celery, django-appconf, flake8, billiard, pytz, kombu, six, mccabe, pep8, pyflakes, amqp, anyjson
  Running setup.py install for billiard
    building '_billiard' extension
    clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -I/usr/local/include -I/usr/local/opt/sqlite/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DHAVE_SEM_OPEN=1 -DHAVE_BROKEN_SEM_GETVALUE=1 -DHAVE_FD_TRANSFER=1 -DHAVE_SEM_TIMEDWAIT=0 -IModules/_billiard -I/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c Modules/_billiard/multiprocessing.c -o build/temp.macosx-10.10-intel-2.7/Modules/_billiard/multiprocessing.o
    clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -I/usr/local/include -I/usr/local/opt/sqlite/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DHAVE_SEM_OPEN=1 -DHAVE_BROKEN_SEM_GETVALUE=1 -DHAVE_FD_TRANSFER=1 -DHAVE_SEM_TIMEDWAIT=0 -IModules/_billiard -I/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c Modules/_billiard/socket_connection.c -o build/temp.macosx-10.10-intel-2.7/Modules/_billiard/socket_connection.o
    clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -I/usr/local/include -I/usr/local/opt/sqlite/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DHAVE_SEM_OPEN=1 -DHAVE_BROKEN_SEM_GETVALUE=1 -DHAVE_FD_TRANSFER=1 -DHAVE_SEM_TIMEDWAIT=0 -IModules/_billiard -I/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c Modules/_billiard/semaphore.c -o build/temp.macosx-10.10-intel-2.7/Modules/_billiard/semaphore.o
    clang -bundle -undefined dynamic_lookup -arch i386 -arch x86_64 -isysroot / -L/usr/local/lib -L/usr/local/opt/sqlite/lib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk build/temp.macosx-10.10-intel-2.7/Modules/_billiard/multiprocessing.o build/temp.macosx-10.10-intel-2.7/Modules/_billiard/socket_connection.o build/temp.macosx-10.10-intel-2.7/Modules/_billiard/semaphore.o -o build/lib.macosx-10.10-intel-2.7/_billiard.so

    warning: no files found matching '*.py' under directory 'Lib'
  Running setup.py install for anyjson

Successfully installed Django celery django-appconf flake8 billiard pytz kombu six mccabe pep8 pyflakes amqp anyjson
Cleaning up...
(django-celery-email)~/S/django-celery-email (_mj_base_task ☡=) python setup.py test
running test
Traceback (most recent call last):
  File "setup.py", line 81, in <module>
    long_description=long_description,
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 29, in run
    execute_from_command_line([__file__, "test"])
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/commands/test.py", line 30, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/base.py", line 381, in run_from_argv
    parser = self.create_parser(argv[0], argv[1])
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/base.py", line 354, in create_parser
    self.add_arguments(parser)
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/core/management/commands/test.py", line 52, in add_arguments
    test_runner_class = get_runner(settings, self.test_runner)
  File "/Users/matt/.virtualenvs/django-celery-email/lib/python2.7/site-packages/django/test/utils.py", line 152, in get_runner
    test_module = __import__(test_module_name, {}, {}, force_str(test_path[-1]))
  File "/Users/matt/Source/django-celery-email/test_project/test_runner.py", line 2, in <module>
    from django.test.simple import DjangoTestSuiteRunner
ImportError: No module named simple
```

Looking at the source, in test_runner.py it's using a deprecated import that has been removed in django 1.8. So it works if you aren't using django 1.8. The requirements.txt doesn't have a django version pinned  so it downloads the latest django which is 1.8 (it probably should pin the requirements if it's using deprecated imports)

```
This module is pending deprecation as of Django 1.6 and will be removed in version 1.8.
```

**Comment by oppianmatt** [comment]
reason why tests are failing with the travis cs is that basestring does not exist in python 3. So really need to use six

**Comment by pmclanahan** [comment]
Sorry I missed that you'd updated this! GH doesn't email me when commits are added :(

Only thing left I think is to add `six` to the `install_requires` in `setup.py`. I've been meaning to do a simple parse of `requirements.txt` in `setup.py` so I can keep requirements updated in one place, but haven't yet.

**Comment by pmclanahan** [comment]
I know we're all short on time. I'll merge this and add six to setup.py. Thanks again.

**Pull request merged**